backend/tracim_backend/lib/search/elasticsearch_search/elasticsearch_search.py

```diff
@@ -281,6 +305,73 @@ def index_contents(self, contents: typing.Iterable[Content]) -> None:
         for content in contents:
             self.index_content(content)
 
+    def index_user(self, user: User) -> None:
+        """Index the given user in the appropriate index."""
+        user_in_context = UserInContext(user, dbsession=self._session, config=self._config)
+
+        rapi = RoleApi(config=self._config, session=self._session, current_user=None)
+        workspace_ids = rapi.get_user_workspaces_ids(user.user_id)
+        capi = ContentApi(config=self._config, session=self._session, current_user=None)
+        try:
+            newest_authored_content_date = capi.get_newest_authored_content(
+                user_in_context.user_id
+            ).created
+        except ContentNotFound:
+            newest_authored_content_date = None
+
+        indexed_user = self.IndexedUser(
+            user_id=user_in_context.user_id,
+            public_name=user_in_context.public_name,
+            username=user_in_context.username,
+            is_deleted=user_in_context.is_deleted,
+            is_active=user_in_context.is_active,
+            has_avatar=user_in_context.has_avatar,
+            has_cover=user_in_context.has_cover,
+            workspace_ids=workspace_ids,
+            custom_properties=user.custom_properties.fields,
+            newest_authored_content_date=newest_authored_content_date,
+        )
+        indexed_user.meta.id = user.user_id
+        user_index_alias = self._get_index_parameters(self.IndexedUser).alias
+        indexed_user.save(
+            using=self.es,
+            index=user_index_alias,
+            request_timeout=self._config.SEARCH__ELASTICSEARCH__REQUEST_TIMEOUT,
+        )
+
+    def delete_user(self, user: User) -> None:
```

Comment:
Not directly related, but as we do "hard" deletion in elasticsearch here for user/workspace, how the deletion of content is handled ? softly/hardly/missing ?

Reply:
After discussion with @grignards soft deletion seems properly handled.

We should probably verify hard deletion too (using tracimcli command, when deleting user).

Reply:
The `is_deleted` case is handled in `on_content_modified`. There is currently no management for database-level delete of contents.